src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasSlicer.java

```diff
@@ -320,44 +875,656 @@ protected boolean isInsideWorkingBound(final AtlasEntity entity)
+     *
+     * @param geometry
+     *            The JTS geometry to slice
+     * @param countryBoundaryPolygons
+     *            All country boundary polygons intersecting it
+     * @return A map of country codes to the portions of JTS geometry inside those country boundary
+     *         polygons
+     */
+    private Map<String, Set<org.locationtech.jts.geom.Geometry>> sliceGeometry(
+            final Geometry geometry,
+            final Set<org.locationtech.jts.geom.Polygon> countryBoundaryPolygons)
+    {
+        final Map<String, Set<org.locationtech.jts.geom.Geometry>> results = new HashMap<>();
+        for (final org.locationtech.jts.geom.Polygon boundaryPolygon : countryBoundaryPolygons)
+        {
+            final String countryCode = CountryBoundaryMap.getGeometryProperty(boundaryPolygon,
+                    ISOCountryTag.KEY);
+            final IntersectionMatrix matrix = geometry.relate(boundaryPolygon);
+            if (matrix.isWithin())
+            {
+                CountryBoundaryMap.setGeometryProperty(geometry, ISOCountryTag.KEY, countryCode);
+                results.clear();
+                results.put(countryCode, new HashSet<>());
+                results.get(countryCode).add(geometry);
+                return results;
+            }
+            else if (matrix.isIntersects())
+            {
+                if (!results.containsKey(countryCode))
+                {
+                    results.put(countryCode, new HashSet<>());
+                }
+                final Geometry clipped = GeometryPrecisionReducer.reduce(
+                        geometry.intersection(boundaryPolygon),
+                        JtsPrecisionManager.getPrecisionModel());
+                if (clipped instanceof GeometryCollection)
+                {
+                    CountryBoundaryMap.geometries((GeometryCollection) clipped)
+                            .filter(this::isSignificantGeometry).forEach(result ->
+                            {
+                                CountryBoundaryMap.setGeometryProperty(result, ISOCountryTag.KEY,
+                                        countryCode);
+                                results.get(countryCode).add(result);
+                            });
+                }
+                else if (isSignificantGeometry(clipped))
+                {
+                    CountryBoundaryMap.setGeometryProperty(clipped, ISOCountryTag.KEY, countryCode);
+                    results.get(countryCode).add(clipped);
+                }
+            }
+        }
+        return results;
+    }
+
+    /**
+     * Slice a Line by converting it to 1d geometry, calculating its slices, and creating the new
+     * sliced Lines. If it belongs to just one country or cannot be sliced, update the ISOCountryTag
+     * appropriately instead
+     *
+     * @param line
+     *            The Line to slice as an Area
+     */
+    private void sliceLine(final Line line)
```

Comment:
This looks similar to sliceArea(), just with lines vs areas; would it benefit from a helper method / class to cover the identical processing?

Reply:
yeah this is a good catch! it originally was one method, but the reality is that I found the forking logic within one method harder to read (and it was impossible to keep at sonar's standards for complexity, too). this approach does lead to some duplication, but long term I think that's easier to maintain.

Reply:
Interesting, so it's a Sonar complexity versus duplication tradeoff.  I trust your maintenance assessment here.